src/fluent-bit.c

```diff
@@ -541,7 +542,7 @@ static void flb_help_plugin(int rc, int format,
 
 static void flb_signal_handler_break_loop(int signal)
 {
-    exit_signal = signal;
+    flb_config_signal_send(config, FLB_CTX_SIGNAL_SHUTDOWN);
```

Comment:
You can't call `flb_pipe_w` from a signal handler. That's why I said you were working on something similar in PR #10275.

Reply:
I think the only real way around using atomic variables will be to add support for signal events to monkey.

Reply:
That's why I tagged you in that other PR. 

I think what he did with the signal queue is correct but we need to do it using the platform agnostic atomic operation wrapper which we need to move from `cmetrics` to `cfl` (I thought that was already done but it seems I was wrong).

I think with that, the signal mask tweak and a bit of elbow grease to ensure that the mechanism is tidy we should be good.

Reply:
Were you able to make any progress on this? Do you need anything from me?